### PR TITLE
Unit test for slack event handler

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: core/jazz_slack-event-handler/test/test.js [hunk 1]
```diff
@@ -1,10 +1,674 @@
...
+      sinon.assert.calledOnce(processEventRecordsStub)
+    })
+  })
+  it("should call getEventProcessStatus after processing Events ", () => {
+    index.handler(event, context, (error, records) => {
+      sinon.assert.calledOnce(getEventProcessStatusStub)
+    })
+  })
+
+  it("should return the record of processed and falied events ", () => {
```
Comment (svidhani) [inline core/jazz_slack-event-handler/test/test.js]: typo

File: core/jazz_slack-event-handler/test/test.js [hunk 2]
```diff
@@ -1,10 +1,674 @@
...
+      uri: "apiEndpoint",
+      method: "GET"
+    }
+  })
+  it("should return payload with values passed by getSvcPayload", () => {
+    const payload = index.getSvcPayload("GET", null, "apiEndpoint", "tempToken")
+    expect(payload.uri).to.eq(svcPayload.uri);
+    expect(payload.headers.authorization).to.eq(svcPayload.token);
+    expect(payload.method).to.eq(svcPayload.method);
+    expect(payload.json).to.eq(undefined);
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 3]
```diff
@@ -1,10 +1,674 @@
...
+const configData = configObj.getConfig(event, context);
+let reqStub, processRequestStub, procesRecordStub, getServiceDetailsStub, processEventStub, notifySlackChannelStub;
+describe("getTokenRequest", function () {
+  it("should return Request token when called", () => {
+    const result = index.getTokenRequest(configData);
+    expect(result.uri).to.eq(configData.SERVICE_API_URL + configData.TOKEN_URL);
+    expect(result.method).to.eq('post');
+  })
+})
+describe("getAuthResponse", () => {
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 4]
```diff
@@ -1,10 +1,674 @@
...
+  "invokeid": "00001-test-000001",
+  "awsRequestId": "00001-test-000001",
+  "invokedFunctionArn": "arn:aws:lambda:region:100000001:function:test10001-test-services-handler-test"
+};
 
-describe('Sample', function() {
-  it('tests handler', function(done) {
+const configData = configObj.getConfig(event, context);
+let reqStub, processRequestStub, procesRecordStub, getServiceDetailsStub, processEventStub, notifySlackChannelStub;
+describe("getTokenRequest", function () {
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 5]
```diff
@@ -1,10 +1,674 @@
...
+        "username": configData.SERVICE_USER,
+        "password": configData.TOKEN_CREDS
+      },
+      rejectUnauthorized: false,
+      transform: (body, response, resolveWithFullResponse) => {
+        return response;
+      }
+    };
+  })
+  it("should give return auth token when called with valid parameters", () => {
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 6]
```diff
@@ -1,10 +1,674 @@
...
+    if (procesRecordStub) { procesRecordStub.restore(); }
+  });
+  it("should resolve all for success scenario from processEventRecord", () => {
+    procesRecordStub = sinon.stub(index, "processRecord").resolves({
+      "status": "succesfully processed Event Rcord"
+    });
+    index.processRecords(event, configData, "temp_auth").then((obj) => {
+      sinon.assert.calledOnce(procesRecordStub)
+      for (let i = 0; i < obj.length; i++) {
+        expect(obj[i].status).to.eq("succesfully processed Event Rcord")
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 7]
```diff
@@ -1,10 +1,674 @@
...
+  let svcPayload;
+  beforeEach(() => {
+    svcPayload = {
+      token: "tempToken",
+      uri: "apiEndpoint",
+      method: "GET"
+    }
+  })
+  it("should return payload with values passed by getSvcPayload", () => {
+    const payload = index.getSvcPayload("GET", null, "apiEndpoint", "tempToken")
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 8]
```diff
@@ -1,10 +1,674 @@
...
+    expect(payload.headers.authorization).to.eq(svcPayload.token);
+    expect(payload.method).to.eq(svcPayload.method);
+    expect(payload.json).to.eq(undefined);
+  })
+  it("should return invalid payload with json data by getSvcPayload", () => {
+    svcPayload.method = "POST"
+    const payload = index.getSvcPayload("GET", null, "apiEndpoint", "tempToken")
+    expect(payload.uri).to.eq(svcPayload.uri);
+    expect(payload.headers.authorization).to.eq(svcPayload.token);
+    expect(payload.method).to.not.eq(svcPayload.method);
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done

File: core/jazz_slack-event-handler/test/test.js [hunk 9]
```diff
@@ -1,10 +1,674 @@
...
+  })
+  it("should return payload with values passed by getSvcPayload", () => {
+    const payload = index.getSvcPayload("GET", null, "apiEndpoint", "tempToken")
+    expect(payload.uri).to.eq(svcPayload.uri);
+    expect(payload.headers.authorization).to.eq(svcPayload.token);
+    expect(payload.method).to.eq(svcPayload.method);
+    expect(payload.json).to.eq(undefined);
+  })
+  it("should return invalid payload with json data by getSvcPayload", () => {
+    svcPayload.method = "POST"
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: Not passing POST, It is assigning for assertion.

File: core/jazz_slack-event-handler/test/test.js (hunk 1 above)
Comment (SiniWilson) [reply core/jazz_slack-event-handler/test/test.js]: done. Thank you

**Comment by suryajak** [comment]
To elaborate further, it would be great if we can have unit test cases for:
1. Slack notification fails
2. Service lookup fails
3. service call fails (non-200 http code and its variations))

**Comment by SiniWilson** [comment]
Updated the PR with including error case scenarios

**Pull request merged**